galaxy/importer/collection.py

```diff
@@ -0,0 +1,154 @@
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+from __future__ import absolute_import
+
+import os
+import tempfile
+import tarfile
+import shutil
+import logging
+
+from galaxy.importer.models import CollectionArtifactManifest, Collection
+from galaxy.importer.utils import readme as readmeutils
+from galaxy.importer import finders as finders_
+from galaxy.importer import loaders
+from galaxy.importer import exceptions as exc
+
+
+default_logger = logging.getLogger(__name__)
+
+
+def import_collection(artifact, logger=None):
+    logger = logger or default_logger
+    with tempfile.TemporaryDirectory() as temp_dir:
+        return CollectionLoader(artifact, temp_dir, logger=logger).load()
+
+
+class CollectionLoader(object):
+    """Loads collection and content info."""
+
+    finders = [
+        finders_.FileSystemFinder,
+    ]
+
+    def __init__(self, artifact, temp_dir, logger=None):
+        self.log = logger or default_logger
+        self.artifact = artifact
+        self.temp_dir = temp_dir
+        self.artifact_path = None
+        self.collection_path = None
+        self.collection_info = None
+        self.contents = None
+        self.readme = None
+
+    def load(self):
+        self._extract_package()
+        self._load_collection_manifest()
+        self._load_collection_readme()
+
+        finder, content_list = self._find_contents()
+        self.contents = list(self._load_contents(content_list))
+
+        self._validate_collection_metadata()
+
+        quality_score = self._get_collection_quality_score()
+
+        return Collection(
+            collection_info=self.collection_info,
+            contents=self.contents,
+            readme=self.readme,
+            quality_score=quality_score,
+        )
+
+    def _extract_package(self):
+        self.artifact_path = os.path.join(
+            self.temp_dir,
+            os.path.basename(self.artifact.file.path))
+        shutil.copy(self.artifact.file.path, self.artifact_path)
+
+        with tarfile.open(self.artifact_path, 'r') as pkg:
+            pkg.extractall(self.temp_dir)
+
+        _, dirs, _ = next(os.walk(self.temp_dir))
+        self.collection_path = os.path.join(self.temp_dir, dirs[0])
+        self.log.info('Extracting collection: {}'.format(dirs[0]))
+        self.log.info(' ')
+
+    def _load_collection_manifest(self):
+        with tarfile.open(self.artifact_path, 'r') as pkg:
+            for member in pkg.getmembers():
+                if member.isfile() and \
+                        member.path.split('/')[-1] == 'MANIFEST.json':
```

Comment:
`os.path.basename`

Reply:
Since you already extracted file, to check if `MANIFEST.json` exists it's sufficient to check `os.path.exists`